lib/virt_autotest/virtual_network_utils.pm

```diff
@@ -44,17 +41,31 @@ sub check_guest_ip {
 
     # ensure guest is still alive
     if (script_output("virsh domstate $guest") eq "running") {
-        script_run "sed -i '/ $guest /d' /etc/hosts";
         my $mac_guest = script_output("virsh domiflist $guest | grep $net | grep -oE \"[[:xdigit:]]{2}(:[[:xdigit:]]{2}){5}\"");
-        my $syslog_cmd = "journalctl --no-pager | grep DHCPACK";
-        script_retry "$syslog_cmd | grep $mac_guest | grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"", delay => 90, retry => 9, timeout => 90;
-        my $gi_guest = script_output("$syslog_cmd | grep $mac_guest | tail -1 | grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"");
-        assert_script_run "echo '$gi_guest $guest # virtualization' >> /etc/hosts";
+        my $gi_guest = '';
+        if (is_alp) {
+            $gi_guest = get_guest_ip_from_vnet_with_mac($mac_guest, $net);
+        } else {
+            my $syslog_cmd = "journalctl --no-pager | grep DHCPACK";
+            script_retry "$syslog_cmd | grep $mac_guest | grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"", delay => 90, retry => 9, timeout => 90;
+            $gi_guest = script_output("$syslog_cmd | grep $mac_guest | tail -1 | grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"");
+        }
+        setup_vm_simple_dns_with_ip($guest, $gi_guest);
         script_retry("nmap $guest -PN -p ssh | grep open", delay => 30, retry => 6, timeout => 60) if ($guest =~ m/sles-11/i);
         die "Ping $guest failed !" if (script_retry("ping -c5 $guest", delay => 30, retry => 6, timeout => 60) ne 0);
     }
 }
 
+sub get_guest_ip_from_vnet_with_mac {
```

Comment:
I think this function name can be shorter, actually "with_mac" is not necessary at all because you just aims to get ip from vnet. So ```get_guest_from_vnet``` is enough.

Reply:
Actually it is also possible to do that with guest name(up to guest setup). Setting it explicitly can avoid name misleading if in future new similar function is needed for whatever reason.